src/hats/loaders/read_hats.py

```diff
@@ -42,7 +44,25 @@ def read_hats(catalog_path: str | Path | UPath) -> Dataset:
             from upath import UPath
             catalog = hats.read_hats(UPath(..., anon=True))
     """
-    catalog_path = file_io.get_upath(catalog_path)
+    path = file_io.get_upath(path)
+    try:
+        collection_properties = CollectionProperties.read_from_dir(path)
+    except FileNotFoundError:
+        return _load_catalog(path)
+    return _load_collection(path, collection_properties)
```

Comment:
```suggestion
    try:
        collection_properties = CollectionProperties.read_from_dir(path)
        return _load_collection(path, collection_properties)
    except FileNotFoundError:
        return _load_catalog(path)
```

Is what you originally had the more pythonic way to do it? My java-dinosaur-brain would want to see the non-exception behavior inside the try/catch.